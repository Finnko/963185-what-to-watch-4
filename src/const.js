const Config = {
  MOVIE_MAX_RATING: 10,
  MOVIE_ACTORS_TO_SHOW: 4,
  SIMILAR_MOVIES_TO_SHOW: 4,
  MOVIES_NUMBER_TO_SHOW: 16,
  MOVIES_NUMBER_LIMIT: 8,
  MOVIES_NUMBER_OFFSET: 8,
  MAX_GENRES_TO_SHOW: 9,
  MOVIE_RATING_MAP: {
    0: `Bad`,
    3: `Normal`,
    5: `Good`,
    8: `Very good`,
    10: `Awesome`
  },
  ERRORS: {
    FETCH_DATA: `Something went wrong, please try again later.`,
  },
  LOADER: {
    SMALL: `small`,
    MEDIUM: `medium`,
  },
  DEFAULT_FILTER: `All genres`,
};

const PathName = {
  ROOT: `/`,
  MOVIE_PAGE: `/movie/`,
};

const TabName = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

const ViewMode = {
  PROMO: {
    MAIN: `MAIN`,
    DETAILS: `DETAILS`
  },
  MOVIE_CARD: {
    WITH_PLAYER: `WITH_PLAYER`,
    IMAGE: `IMAGE`,
  }
};

export {
  Config,
  PathName,
  TabName,
  ViewMode,
};
