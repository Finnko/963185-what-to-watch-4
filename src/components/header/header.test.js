import React from 'react';
import renderer from 'react-test-renderer';
import {createMemoryHistory} from 'history';
import {Link, Router} from 'react-router-dom';
import {Provider} from 'react-redux';
import NameSpace from '../../store/name-space';
import configureStore from 'redux-mock-store';
import Header from './header';
import {PathName} from '../../const';


const mockUser = {
  id: 1,
  email: `Oliver.conner@gmail.com`,
  name: `Oliver.conner`,
  avatarUrl: `img/1.png`
};

const mockStore = configureStore([]);

const store = mockStore({
  [NameSpace.USER]: {
    user: mockUser,
    authStatus: `NO_AUTH`,
  }
});

describe(`Header component render correctly`, () => {
  const history = createMemoryHistory();

  it(`Should Header component render correctly without children`, () => {
    const props = {
      className: ``,
      children: [],
    };

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <Header {...props}/>
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should Header component render correctly with children`, () => {
    const props = {
      className: ``,
    };

    const tree = renderer
      .create(
          <Provider store={store}>
            <Router history={history}>
              <Header {...props}>
                <nav className="breadcrumbs">
                  <ul className="breadcrumbs__list">
                    <li className="breadcrumbs__item">
                      Main
                    </li>
                    <li className="breadcrumbs__item">
                      Details
                    </li>
                  </ul>
                </nav>
              </Header>
            </Router>
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
