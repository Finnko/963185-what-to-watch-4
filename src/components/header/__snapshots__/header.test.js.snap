// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component render correctly Should Header component render correctly with children 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <nav
    className="breadcrumbs"
  >
    <ul
      className="breadcrumbs__list"
    >
      <li
        className="breadcrumbs__item"
      >
        Main
      </li>
      <li
        className="breadcrumbs__item"
      >
        Details
      </li>
    </ul>
  </nav>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign In
    </a>
  </div>
</header>
`;

exports[`Header component render correctly Should Header component render correctly without children 1`] = `
<header
  className="page-header "
>
  <div
    className="logo"
  >
    <a
      className="logo__link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="logo__letter logo__letter--1"
      >
        W
      </span>
      <span
        className="logo__letter logo__letter--2"
      >
        T
      </span>
      <span
        className="logo__letter logo__letter--3"
      >
        W
      </span>
    </a>
  </div>
  <div
    className="user-block"
  >
    <a
      className="user-block__link"
      href="/login"
      onClick={[Function]}
    >
      Sign In
    </a>
  </div>
</header>
`;
